BackEnd/EdgeWeights/HTTPRequest.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  HTTPRequest.swift
+//
+//  "https://maps.googleapis.com/maps/api/geocode/json?address=1600+Amphitheatre+Parkway,+Mountain+View,+CA&key=AIzaSyBph9FtyDmDvZqBFuh00noX-kfgI4bhU7I"
+//
+//  Created by Guest Code User on 3/2/19.
+//
+
+import Foundation
+
+let apiKey = "AIzaSyCrjARni6M94n5tNwj6SnsX-qdRNKEWVvk"
+func getETA(origin: String, destination: String) -> Int {
+    var result: String = "" 
+    let modOrigin = origin.replacingOccurrences(of: " ", with: "+")
+    let modDestination = destination.replacingOccurrences(of: " ", with: "+")
+ 
+    let url = URL(string: "https://maps.googleapis.com/maps/api/distancematrix/json?units=imperial&origins=" + modOrigin + "&destinations=" + modDestination + "&key=" + apiKey)!
+    
+    let task:URLSessionDataTask = URLSession.shared.dataTask(with: url) {(data, response, error) in
+        guard let data = data else {return}
+        result = String(data: data, encoding: .utf8)!
+    }
+    
+    while(result == "") {
+        task.resume()
+    }
+    task.cancel()
+    
+    return readString(result)    
+}
+
+func readString(_ input: String) -> Int {
```

Comment:
Can you put some comments in here?

This function appears to me as a complicated string parser for wherever the ETA field is in the returned string from Google Maps API.

If it works, it works, but one thing you could try to make the logic clearer:
1) Is it a JSON string that's returned (from Google Maps API)? If so, you could try adding this to `func getETA` on line 21 to parse it as JSON instead of as just a UTF
`let jsonResponse = try JSONSerialization.jsonObject(with: data, options: [])`
Check out this article: https://medium.com/@nimjea/json-parsing-in-swift-2498099b78f